library/core/src/sync/atomic.rs

```diff
@@ -3149,13 +3137,13 @@ macro_rules! atomic_int {
             #[$cfg_cas]
             #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
             pub fn fetch_update<F>(&self,
-                                   set_order: Ordering,
-                                   fetch_order: Ordering,
+                                   success: Ordering,
+                                   failure: Ordering,
                                    mut f: F) -> Result<$int_type, $int_type>
             where F: FnMut($int_type) -> Option<$int_type> {
-                let mut prev = self.load(fetch_order);
+                let mut prev = self.load(failure);
                 while let Some(next) = f(prev) {
-                    match self.compare_exchange_weak(prev, next, set_order, fetch_order) {
+                    match self.compare_exchange_weak(prev, next, success, failure) {
                         x @ Ok(_) => return x,
                         Err(next_prev) => prev = next_prev
                     }
```

Comment:
Please make sure to update `update` and `try_update` as well once this is closer to ready https://github.com/rust-lang/rust/issues/135894